### PR TITLE
chore(developer): honor prompt to upgrade in kmc

### LINKED ISSUE #10162: chore(developer): honor setting in #10146 in kmc
> Note: this should probably honour the setting in #10146.

Blocked by #10161.

_Originally posted by @mcdurdin in https://github.com/keymanapp/keyman/issues/10147#issuecomment-1843779195_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10162.

@keymanapp-test-bot skip

### CODE REVIEW

**Review by jahorton — APPROVED** [review]

File: developer/src/kmc/src/commands/buildClasses/BuildProject.ts [hunk 1]
```diff
@@ -43,7 +44,9 @@ class ProjectBuilder {
 
     // Give a hint if the project is v1.0
     if(this.project.options.version != '2.0') {
-      this.callbacks.reportMessage(InfrastructureMessages.Hint_ProjectIsVersion10());
+      if(getOption("prompt to upgrade projects", true)) {
```
Comment (darcywong00) [inline developer/src/kmc/src/commands/buildClasses/BuildProject.ts:47]: Could you combine the if clauses?

File: developer/src/kmc/test/test-project-build.ts [hunk 2]
```diff
@@ -4,11 +4,13 @@ import 'mocha';
 import { BuildProject } from '../src/commands/buildClasses/BuildProject.js';
 import { makePathToFixture } from './helpers/index.js';
 import { InfrastructureMessages } from '../src/messages/infrastructureMessages.js';
+import { clearOptions } from '@keymanapp/developer-utils';
 
 const callbacks = new TestCompilerCallbacks();
 
 describe('BuildProject', function () {
   it('should build a keyboard project', async function() {
+    clearOptions();
```
Comment (darcywong00) [inline developer/src/kmc/test/test-project-build.ts:13]: Would this be consistent with the other test files?

```suggestion
  beforeEach(clearOptions);
  
  it('should build a keyboard project', async function() {
```

**Review by darcywong00 — APPROVED** [review]
lgtm

File: developer/src/kmc/src/commands/buildClasses/BuildProject.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc/src/commands/buildClasses/BuildProject.ts:47]: Yeah but I don't think it's worth it :grin:

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.237-alpha](https://keyman.com/downloads/releases/alpha/17.0.237)

**Pull request merged**